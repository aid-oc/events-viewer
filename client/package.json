{
  "name": "feast-events-client",
  "version": "0.1.0",
  "private": true,
  "proxy": "http://localhost:5000",
  "dependencies": {
    "@apollo/client": "^3.1.3",
    "@fortawesome/fontawesome-svg-core": "^1.2.30",
    "@fortawesome/free-solid-svg-icons": "^5.14.0",
    "@fortawesome/react-fontawesome": "^0.1.11",
    "graphql": "^15.3.0",
    "prop-types": "^15.7.2",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "^3.4.1",
    "styled-components": "^5.1.1"
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.3.2",
    "@testing-library/user-event": "^7.1.2",
    "eslint-config-airbnb": "^18.2.0",
    "eslint-plugin-import": "^2.22.0",
    "eslint-plugin-jsx-a11y": "^6.3.1",
    "eslint-plugin-react": "^7.20.6",
    "eslint-plugin-react-hooks": "^4.1.0"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
