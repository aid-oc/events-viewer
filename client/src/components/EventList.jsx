import React from 'react';
import PropTypes from 'prop-types';
import Event from './Event';
import { eventShape, posterShape } from './eventsShape';

const EventList = ({ events }) => <ul>{events.map((event) => <Event {...event} />)}</ul>;

EventList.propTypes = {
  events: PropTypes.arrayOf(PropTypes.shape({
    event: eventShape,
    poster: posterShape,
  })).isRequired,
};

export default EventList;
