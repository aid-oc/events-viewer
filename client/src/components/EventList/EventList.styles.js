import styled from 'styled-components';

const EventItem = styled.li`
    display: flex;
    flex-direction: column;
    margin-bottom: 2rem;
    border-radius: 1rem;
    padding: 1rem;
    text-align: left;
    max-width: 20rem;

    -webkit-box-shadow :0px 3px 15px rgba(0,0,0,0.2);
    -moz-box-shadow: 0px 3px 15px rgba(0,0,0,0.2);
    box-shadow: 0px 3px 15px rgba(0,0,0,0.2);
    
    @media (min-width: 768px)  {
        justify-content: space-around;
        margin: 2rem;
    }
`;

const EventItems = styled.ul`
    display: flex;
    flex-direction: column;
    padding: 0;
    @media (min-width: 768px)  {
        flex-direction: row;
    }
`;

export { EventItem, EventItems };
