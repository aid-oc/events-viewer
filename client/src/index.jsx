import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import { ApolloProvider } from '@apollo/client';
import App from './App';
import createClient from './graph/createClient';

ReactDOM.render(
  <React.StrictMode>
    <ApolloProvider client={createClient()}>
      <App />
    </ApolloProvider>
  </React.StrictMode>,
  document.getElementById('root'),
);

if (module.hot) {
  module.hot.accept();
}
