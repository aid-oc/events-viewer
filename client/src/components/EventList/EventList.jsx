import React from 'react';
import PropTypes from 'prop-types';
import Event from '../EventEntry/Event';
import Poster from '../EventEntry/Poster';
import { eventShape, posterShape } from '../eventsShape';
import { EventItem, EventItems } from './EventList.styles';

const EventList = ({ events }) => (
  <EventItems>
    {events.map((eventEntry) => (
      <EventItem key={`event-${eventEntry.poster.name}-${eventEntry.event.date}`}>
        <Poster poster={eventEntry.poster} />
        <Event event={eventEntry.event} />
      </EventItem>
    ))}
  </EventItems>
);

EventList.propTypes = {
  events: PropTypes.arrayOf(PropTypes.shape({
    event: eventShape,
    poster: posterShape,
  })).isRequired,
};

export default EventList;
