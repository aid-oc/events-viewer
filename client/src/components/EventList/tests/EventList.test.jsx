import React from 'react';
import {
  render,
} from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import EventList from '../EventList';

const createEvent = (id) => ({
  event: {
    id,
    description: 'eventDescription',
    date: 'eventDate',
    guestCount: 1,
    type: 'eventType',
    postcode: 'NG1 4GL',
    address: 'Example Avenue',
    cancelled: false,
    budget: 500,
    dietary: ['vegetarian'],
  },
  poster: {
    email: 'test@feast.com',
    image: 'exampleImageUri',
    name: `${id}-exampleName`,
    phone: '01213590124',
  },
});

const exampleEvents = [createEvent('1'), createEvent('2')];

describe('<EventList />', () => {
  it('Should match snapshot, correctly displaying two events', () => {
    const { asFragment } = render(
      <MockedProvider mocks={[]}>
        <EventList events={exampleEvents} />
      </MockedProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
