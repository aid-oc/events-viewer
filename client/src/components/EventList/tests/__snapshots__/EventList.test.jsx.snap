// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventList /> Should match snapshot, correctly displaying two events 1`] = `
<DocumentFragment>
  <ul
    class="sc-AxhUy gQZmaM"
  >
    <li
      class="sc-AxhCb iDpQiD"
    >
      <div
        class="sc-AxiKw kgbkAL"
      >
        <img
          alt="1-exampleName"
          src="exampleImageUri"
        />
        <h4>
          <b>
            1-exampleName
          </b>
        </h4>
        <span>
          <a
            href="mailto: test@feast.com"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-envelope fa-w-16 "
              data-icon="envelope"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                fill="currentColor"
              />
            </svg>
          </a>
           test@feast.com
        </span>
        <p>
          01213590124
        </p>
      </div>
      <div
        class="sc-AxjAm jfXUMh"
      >
        <p>
          <b>
            Date: 
          </b>
          eventDate
        </p>
        <p>
          <b>
            Budget: 
          </b>
          £500
        </p>
        <p>
          <b>
            Guests: 
          </b>
          1
        </p>
        <p>
          <b>
            Event Type: 
          </b>
          eventType
        </p>
        <p>
          <b>
            Description: 
          </b>
          eventDescription
        </p>
        <p>
          <b>
            Dietary: 
          </b>
          vegetarian
        </p>
        <p>
          <b>
            Location: 
          </b>
          NG1 4GL, Example Avenue
        </p>
      </div>
    </li>
    <li
      class="sc-AxhCb iDpQiD"
    >
      <div
        class="sc-AxiKw kgbkAL"
      >
        <img
          alt="2-exampleName"
          src="exampleImageUri"
        />
        <h4>
          <b>
            2-exampleName
          </b>
        </h4>
        <span>
          <a
            href="mailto: test@feast.com"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-envelope fa-w-16 "
              data-icon="envelope"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                fill="currentColor"
              />
            </svg>
          </a>
           test@feast.com
        </span>
        <p>
          01213590124
        </p>
      </div>
      <div
        class="sc-AxjAm jfXUMh"
      >
        <p>
          <b>
            Date: 
          </b>
          eventDate
        </p>
        <p>
          <b>
            Budget: 
          </b>
          £500
        </p>
        <p>
          <b>
            Guests: 
          </b>
          1
        </p>
        <p>
          <b>
            Event Type: 
          </b>
          eventType
        </p>
        <p>
          <b>
            Description: 
          </b>
          eventDescription
        </p>
        <p>
          <b>
            Dietary: 
          </b>
          vegetarian
        </p>
        <p>
          <b>
            Location: 
          </b>
          NG1 4GL, Example Avenue
        </p>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
